lib/_stream_writable.js

```diff
@@ -293,22 +293,16 @@ function doWrite(stream, state, writev, len, chunk, encoding, cb) {
 }
 
 function onwriteError(stream, state, sync, er, cb) {
+  --state.pendingcb;
```

Comment:
I'm not sure if it's correct to decrement the counter upfront.  Maybe e.g. @chrisdickinson can chime in?